xtask/src/main.rs

```diff
@@ -752,18 +762,26 @@ fn lint_packages(workspace: &Path, args: LintPackagesArgs) -> Result<()> {
                 Package::Examples | Package::HilTest | Package::QaTest => {}
 
                 // By default, no `clippy` arguments are required:
-                _ => lint_package(&path, &[], args.fix)?,
+                _ => lint_package(chip, &path, &[], args.fix)?,
             }
         }
     }
 
     Ok(())
 }
 
-fn lint_package(path: &Path, args: &[&str], fix: bool) -> Result<()> {
+fn lint_package(chip: &Chip, path: &Path, args: &[&str], fix: bool) -> Result<()> {
     log::info!("Linting package: {}", path.display());
 
-    let mut builder = CargoArgsBuilder::default().subcommand("clippy");
+    let builder = CargoArgsBuilder::default().subcommand("clippy");
+
+    let mut builder = if chip.is_xtensa() {
+        // We only overwrite Xtensas so that externally set nightly/stable toolchains
+        // are not overwritten.
+        builder.toolchain("esp")
+    } else {
+        builder
+    };
```

Comment:
Doesn't this mean that we will be linting with different compiler versions depending on which chip is being targeted? The `esp` toolchain is based on `nightly`, which will have a different set of lint rules than the `stable` toolchain used for RISC-V. Perhaps we should always use `esp`? Or am I missing something?

Reply:
It's a bit weird, because Xtensa is technically nightly, but a nightly that is built from the latest stable. But this isn't important here, I think.

We need to use the esp toolchain to lint Xtensa MCUs, otherwise the xtask just fails with a lot of "unknown cpu" sadness. So this part is essentially a QoL change so that we don't __need__ to always invoke it using `cargo +esp xtask`.

This is not changing the invocation for RISC-V chips, so that the CI can still have a choice between the stable/nightly upstream compilers.